sound/soc/intel/boards/bdw-rt5677.c

```diff
@@ -71,6 +71,7 @@ static const struct snd_soc_dapm_route bdw_rt5677_map[] = {
 	{"DMIC L2", NULL, "Local DMICs"},
 	{"DMIC R2", NULL, "Local DMICs"},
 
+#if !IS_ENABLED(CONFIG_SND_SOC_SOF_BROADWELL)
```

Comment:
@plbossart are you possibly missing the virtual widgets for the other three "AIF1 Capture", "AIF1 Playback" and "DSP Buffer"?

Reply:
AIF1 playback/capture are on the codec side, so are defined by the codec driver.

I have no idea what the 'DSP Buffer' is

Reply:
DSP Buffer is for the codecs internal DSP power graph.

Reply:
@plbossart That was strange, I redo the test without this patch on samus with the latest tplg from https://github.com/thesofproject/sof/blob/c68605403c52ac43ac1b97dec780b9407026ec7f/tools/topology/sof-bdw-codec.m4 and it works without error.

```
dnl CODEC is defined and will be expanded, need to undefine it before use
undefine(`CODEC')
VIRTUAL_WIDGET(SSP0 CODEC OUT, output, 0)
VIRTUAL_WIDGET(SSP0 CODEC IN, input, 1)
VIRTUAL_WIDGET(DSP Capture, input, 2)
```

Reply:
@xiulipan thanks for testing, indeed this is not needed, I was using an old topology file :-(

Fixed now